src/chain.h

```diff
@@ -191,10 +191,10 @@ static const BlockStatus BLOCK_VALID_CONSENSUS = BLOCK_VALID_SCRIPTS;
  */
 class CBlockIndex
 {
-public:
-    //! pointer to the hash of the block, if any. Memory is owned by this CBlockIndex
-    const uint256* phashBlock;
+    //! The hash of the block, if it has been calculated
+    std::optional<uint256> optHashBlock;
```

Comment:
This is going to increase memory usage, because the block hash will be duplicated in memory and every `CBlockIndex` will now store 33 bytes instead of 8. For now this is only around 44 MiB so it's probably fine.

Separately, this is a deviation from upstream that we should think through.

Reply:
I don't necessarily think we should do this change; it was just the simplest way to avoid memory handling issues when setting the hash from the block header.

Reply:
This has been undone.